### ISSUE TITLE
[telemetry] Per-page span and duration histogram in AbstractWebPage.getContent

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#42]
## Goal

Emit a per-page ph-telemetry span and duration histogram from `AbstractWebPage.getContent (...)` in `ph-oton-uicore`.

This is the only place in ph-oton where a **bounded, meaningful per-screen dimension** exists — the page ID. It answers "which screen of my application is slow?", which none of the other instrumentation points can.

## Where

`ph-oton-uicore/src/main/java/com/helger/photon/uicore/page/AbstractWebPage.java`

```java
public final void getContent (@NonNull final WPECTYPE aWPEC)
```

The method is `final`, so every page in every downstream application (Bootstrap 4/5 bindings, phoss-smp, peppol-practical, …) passes through it. The body dispatches to `isValidToDisplayPage`, `beforeFillContent`, `fillContent`, `afterFillContent` and `onInvalidToDisplayPage`.

The method declares no checked exception, so `Telemetry.withSpanVoid (...)` is sufficient — no throwing variant needed.

## Cardinality

The page ID comes from `AbstractPage.getID ()`. It is a registered, statically known menu item ID — bounded by the number of screens in the application, typically tens. **This is safe as a metric attribute** and is the whole point of this issue.

Do not add the locale, the logged-in user, or any request parameter to the metric attributes. The locale from `aWPEC.getDisplayLocale ()` is acceptable as a **span** attribute.

## What to emit

New `com.helger.photon.uicore.CUICoreTelemetry`:

| Constant | Value |
|---|---|
| `SPAN_PAGE_CONTENT` | `photon.webpage.content` |
| `METRIC_PAGE_CONTENT` | `photon.webpage.rendered` |
| `METRIC_PAGE_CONTENT_DURATION` | `photon.webpage.duration` |
| `ATTR_PAGE_ID` | `photon.webpage.id` (bounded — metric + span) |
| `ATTR_PAGE_DISPLAYED` | `photon.webpage.displayed` (`false` when `isValidToDisplayPage` rejected) |
| `ATTR_PAGE_LOCALE` | `photon.webpage.locale` (**span only**) |
| `UNIT_PAGE` | `{page}` |
| `UNIT_MILLIS` | `ms` |

Plus a `WebPageMetrics` instrument holder and a package-private `WebPageTelemetry` emitter, mirroring `ph-oton-mgrs/.../longrun/`.

**Span** `photon.webpage.content`, kind `INTERNAL`, attributes page ID + locale + displayed flag. It nests inside the `photon.page.request` span from the UI rendering issue when both are present.

**Counter** `photon.webpage.rendered`, attributes page ID + displayed.
**Histogram** `photon.webpage.duration` in ms, attribute page ID.

## Note on `isValidToDisplayPage`

When `isValidToDisplayPage (aWPEC)` returns invalid, `onInvalidToDisplayPage (aWPEC)` runs instead of the fill-content chain. Both branches must still emit the span and the counter, distinguished by `photon.webpage.displayed`. A page that is silently never displayed (e.g. a permission problem) is exactly the kind of thing this instrument should surface.

## POM

`ph-oton-uicore` gets `ph-telemetry` transitively via `ph-oton-core` → `ph-oton-mgrs`, but declare it directly since it is used directly (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Span, counter and histogram emitted for both the displayed and the not-displayed branch.
- [ ] Page ID is the only metric dimension; locale is span-only.
- [ ] Unit test with recording SPIs installed via `Telemetry.install (...)` / `TelemetryMetrics.install (...)` and reset to `null` in `@After`.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.5.1**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-uicore/.../uicore/CUICoreTelemetry.java` | new, public - span / event / metric / attribute names |
| `ph-oton-uicore/.../uicore/page/WebPageMetrics.java` | new, public - counter `photon.webpage.rendered`, histogram `photon.webpage.duration` |
| `ph-oton-uicore/.../uicore/page/WebPageTelemetry.java` | new, package-private emitter |
| `ph-oton-uicore/.../uicore/page/AbstractWebPage.java` | modified |
| `ph-oton-uicore/pom.xml` | `com.helger.telemetry:ph-telemetry` added directly (no `<version>`, BOM) |
| `ph-oton-uicore/src/test/.../WebPageTelemetryTest.java` | new, 3 tests |

### Deviation from the issue text: manual span instead of `withSpanVoid`

The issue expected `Telemetry.withSpanVoid (...)`, because `getContent (...)` declares no checked exception. That does not survive contact with `AbstractWebPageForm`: `fillContent (...)` routinely ends in `aWPEC.postRedirectGet (...)`, which throws `ForcedRedirectException` - a `RuntimeException`. `withSpanVoid` unconditionally records any escaping `RuntimeException` and sets the span status to `ERROR`, so **every successful form submission would have produced a failed span**.

The span is therefore managed with `Telemetry.startSpan (...)` in a try-with-resources plus an explicit `catch (ForcedRedirectException)` that adds the span event `photon.webpage.forcedredirect` and sets the status to OK. This mirrors what `XServletHandlerToSimpleHandler` does with the same exception. New constant `CUICoreTelemetry.EVENT_FORCED_REDIRECT`.

A pleasant side effect: without a lambda there is no need for a mutable box - `boolean bDisplayed` is a plain local read from the `finally` block.

### Emission

* Span `photon.webpage.content` (`INTERNAL`), attributes `photon.webpage.id`, `photon.webpage.locale` (span only) and `photon.webpage.displayed`.
* Counter `photon.webpage.rendered` with `photon.webpage.id` + `photon.webpage.displayed`.
* Histogram `photon.webpage.duration` with `photon.webpage.id` only.

Both branches emit span, counter and histogram, distinguished by `photon.webpage.displayed` - a page that is silently never displayed because `isValidToDisplayPage (...)` rejected it is exactly as visible as one that renders, which was the point of the issue.

### Cardinality

`getID ()` from `AbstractPage` is the only metric dimension. `aWPEC.getDisplayLocale ().toString ()` is a span attribute only; `WebPageTelemetryTest.testDisplayedPage` asserts that the counter attribute map has exactly 2 entries and no locale, and that the histogram attribute map has exactly 1.

### Verification

`WebPageTelemetryTest` drives a real `AbstractWebPage` through a real `WebPageExecutionContext` (`MenuTree` → `SimpleWebExecutionContext` → `LayoutExecutionContext`) and covers the displayed branch, the not-displayed branch and a `fillContent` that ends in a Post-Redirect-Get.

Full reactor `mvn install` on all 23 modules: BUILD SUCCESS, no test failures. `README.md` news bullet added under the new `v10.5.1 - work in progress` section.
